internal/controller/cluster_controller.go

```diff
@@ -95,17 +59,17 @@ func (controller ControllerConfig) ClusterHandler(c *gin.Context) {
 // @Router       /v1/clusters/ [get]
 // @Security BasicAuth
 func (controller ControllerConfig) ClusterListHandler(c *gin.Context) {
-	var clusterList v1.ClusterList
+	var clusterListResponse v1.ClusterList
 
-	clusterApiListCR, err := controller.K8sInstance.ListClusters()
+	clusterList, err := kaas.ListClusters(controller.K8sInstance)
 	if err != nil {
-		log.Printf("[ClusterListHandler] Error getting clusterAPI CR: %s", err.Error())
+		log.Printf("[ClusterListHandler] Error getting cluster List: %s", err.Error())
 		clientErr, ok := err.(*clientError.ClientError)
 		if !ok {
 			clientError.ErrorHandler(c, err, "Internal Server Error", http.StatusInternalServerError)
 		} else {
 			if clientErr.ErrorMessage == clientError.ResourceNotFound {
-				clientError.ErrorHandler(c, err, "Cluster not found", http.StatusNotFound)
+				clientError.ErrorHandler(c, err, "No Clusters were found", http.StatusNotFound)
 			} else if clientErr.ErrorMessage == clientError.EmptyResponse {
 				clientError.ErrorHandler(c, err, "No clusters were found", http.StatusNotFound)
```

Comment:
I will changed to use lower case in all matches, because the ErrorHandler is an end-user friendly response message. The end-user does not know the Cluster struct or custom resource